packages/functional-tests/pages/products/index.ts

```diff
@@ -13,7 +13,6 @@ export class SubscribePage extends BaseLayout {
     await frame.fill('.InputElement[name=cvc]', '333');
     await frame.fill('.InputElement[name=postal]', '66666');
     await this.page.check('input[type=checkbox]');
-    await this.page.click('button[type=submit]');
```

Comment:
I might be a bit confused.
Why remove the auto-submit here? (I understand it for the `setFailedCreditCardInfo()` method below, but wasn't sure if it was for consistency, or something else).

Reply:
I created a function `clickPayNow()` for the submit action for being able to use it without filling in the card details in the coupon.spec.ts tests. So I thought to remove the step from `setFailedCreditCardInfo()` and `setCreditCardInfo()` so to not be repetitive.

Reply:
Yeah, looking back, I prefer your consistency approach of having both methods do the same thing (and not submit the form).

Alternatively, we could change the `setCreditCardInfo()` method to accept an object of values to fill, and then just have constants for `validCard` and `invalidCard`; then we only need one method.

```js
const validCard = {
    cardnumber: '4242424242424242',
    expdate: '555',
    cvc: '333',
    postal: '66666',
};
const invalidCard = {
    cardnumber: '4000000000000341',
    expdate: '666',
    cvc: '444',
    postal: '77777',
};

  async setCreditCardInfo(cardInfo = validCard) {
    const frame = this.page.frame({ url: /elements-inner-card/ });
    // await frame.fill('.InputElement[name=cardnumber]', ''); // Needed?
    await frame.fill('.InputElement[name=cardnumber]', cardInfo.cardnumber);
    await frame.fill('.InputElement[name=exp-date]', cardInfo.expdate);
    await frame.fill('.InputElement[name=cvc]', cardInfo.cvc);
    await frame.fill('.InputElement[name=postal]', cardInfo.postal);
    await this.page.check('input[type=checkbox]');
}
```

Then below in tests

```diff
-await subscribe.setFailedCreditCardInfo();
+await subscribe.setCreditCardInfo(invalidCard);
```

&hellip; but that's another refactor for another time.

Reply:
Yes, I will definitely keep this in mind for the next PR. Thanks!